src/tools/rbd/action/ImageMeta.cc

```diff
@@ -94,6 +94,24 @@ static int do_metadata_set(librbd::Image& image, const char *key,
   return r;
 }
 
+static int do_matadata_keycheck(librbd::Image& image, const char *key){
+    std::map<std::string, bufferlist> pairs;
+    int r;
+    r = image.metadata_list("", 0, &pairs);
```

Comment:
now i'm using cls_client::metadata_get() function to test whether the key exists in operation.cc::metadata_remove，is that what you mean?